lib/domain/services/background_sync_service.dart

```diff
@@ -0,0 +1,692 @@
+    );
+    return 0;
+  }
+  await _notifyNewMessages(
+    groupId: groupId,
+    activePubkey: activePubkey,
+    newMessages: newMessages,
+    logger: logger,
+  );
+  await _updateCheckpointWithNewMessages(
+    activePubkey: activePubkey,
+    groupId: groupId,
+    newMessages: newMessages,
+  );
+  return newMessages.length;
+}
+
+Future<void> _notifyNewMessages({
+  required String groupId,
+  required String activePubkey,
+  required List<ChatMessage> newMessages,
+  required Logger logger,
+}) async {
+  final String groupDisplayName = await BackgroundSyncService._getGroupDisplayName(
+    groupId,
+    activePubkey,
+  );
+  for (final ChatMessage message in newMessages) {
+    try {
+      await NotificationService.showMessageNotification(
+        id: await NotificationIdService.getIdFor(
+          key: '${BackgroundSyncService._notificationTypeNewMessage}:$groupId:${message.id}',
+        ),
+        title: groupDisplayName,
+        body: message.content,
+        payload: jsonEncode({
+          'type': BackgroundSyncService._notificationTypeNewMessage,
+          'groupId': groupId,
+          'messageId': message.id,
+          'sender': message.pubkey,
+        }),
+      );
+    } catch (e) {
+      logger.warning('Show notification for message ${message.id}: $e');
+    }
+  }
+}
+
+Future<void> _updateCheckpointNoNewMessages({
+  required String activePubkey,
+  required String groupId,
+  required DateTime? lastSyncTime,
+}) async {
+  final DateTime now = DateTime.now();
+  final DateTime bufferCutoff = now.subtract(BackgroundSyncService._messageFilterBufferSeconds);
+  if (lastSyncTime == null || lastSyncTime.isBefore(bufferCutoff)) {
+    await BackgroundSyncService._setLastSyncTime(
+      activePubkey: activePubkey,
+      groupId: groupId,
+      time: bufferCutoff,
+    );
+  }
+}
+
+Future<void> _updateCheckpointWithNewMessages({
+  required String activePubkey,
+  required String groupId,
+  required List<ChatMessage> newMessages,
+}) async {
+  final DateTime latestProcessed = newMessages
+      .map((m) => m.createdAt)
+      .reduce((a, b) => a.isAfter(b) ? a : b);
+  await BackgroundSyncService._setLastSyncTime(
+    activePubkey: activePubkey,
+    groupId: groupId,
+    time: latestProcessed,
+  );
+}
+
+Future<bool> _handleInvitesSync() async {
+  final logger = Logger('InvitesSyncTask');
+
+  try {
+    logger.info('Starting invites sync background task');
+
+    final activePubkey = await AccountSecureStorageService.getActivePubkey();
+    if (activePubkey == null) {
+      logger.info('No active account found, skipping invites sync');
+      return true;
+    }
+
+    // Use checkpoint tracking for invites like we do for messages
+    final DateTime? lastSyncTime = await BackgroundSyncService._getLastSyncTime(
+      activePubkey: activePubkey,
+      groupId: 'invites',
+    );
+    final now = DateTime.now();
+    final bufferCutoff = now.subtract(BackgroundSyncService._messageFilterBufferSeconds);
```

Comment:
I was confused by this variable name, I din't realize that it was the most recent time to include at first... Maybe `mostRecentTimeToInclude` or `upToTime` would help?